api/provider/aws/load_balancer_delete.go

```diff
@@ -44,6 +45,19 @@ func (l *LoadBalancerProvider) Delete(loadBalancerID string) error {
 		}
 	}
 
+	// Confirm Delete
+	describeFN := func(id string) error {
+		_, err := describeLoadBalancer(l.AWS.ELB, id)
+		if serverError, ok := err.(*errors.ServerError); ok && serverError.Code == "LoadBalancerDoesNotExist" {
```

Comment:
We need to operate on a lower level: the problem here is the loadbalancer's security group, so we need to make calls to the aws api to verify that the security group is deleted

Reply:
I would add that additional items need to be checked. Keep in mind that when you create a load balancer you also create an IAM Role and a Role Policy. We should probably check that these items are removed as well.